dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ChatCompletionStreamingResult.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.AzureSdk;
+
+internal sealed class ChatStreamingResult : IChatStreamingResult, ITextCompletionStreamingResult
+{
+    private readonly StreamingChatChoice _choice;
+
+    public ChatStreamingResult(StreamingChatChoice choice)
+    {
+        this._choice = choice;
```

Comment:
Should we check the parameter for null? because if it's null neither of the class methods are able to fulfill their contract.